// @lekoarts/gatsby-theme-minimal-blog/src/components/listing.tsx
/** @jsx jsx */
import BlogListItem from "./blog-list-item"
import { Box } from "@theme-ui/components"  
import { jsx, Heading, Link as TLink } from "theme-ui"
type ListingProps = {
  posts: {
    slug: string
    title: string
    date: string
    excerpt: string
    description: string
    timeToRead?: number
    tags?: {
      name: string
      slug: string
    }[]
  }[]
  className?: string
  showTags?: boolean
}

const Listing = ({ posts, className = ``, showTags = true }: ListingProps) => (
  <section sx={{ mb: [5, 6, 7] }} className={className}>
    {posts.map((post) => (
      <BlogListItem key={post.slug} post={post} showTags={showTags} />
    ))}
            <Box mb={4}>
            <a href="/resources" sx={{textDecoration:"none"}}>
              <TLink as="resources" to="/resources" sx={{ fontSize: [1, 2, 3], color: `text` }}>
                Resources
              </TLink>
              <p sx={{ color: `secondary`, mt: 1, a: { color: `secondary` }, fontSize: [1, 1, 2] }} style={{padding:"0px",margin:"0px"}}>
                <p style={{padding:"0px",margin:"0px"}}>
                  This year, Roboknights released its learning resources to help every budding roboticists to gather knowledge
                </p>
                
                
              </p>
            </a>
            </Box>
  </section>
)

export default Listing
